universe/Ship.cpp

```diff
@@ -438,7 +438,7 @@ float Ship::TotalWeaponsShipDamage(float shield_DR, bool include_fighters) const
 }
 
 namespace {
-    std::vector<float> WeaponShipDamageImpl(const Ship* ship, const ShipDesign* design,
+    std::vector<float> WeaponShipDamageImpl(const std::shared_ptr<Ship> ship, const ShipDesign* design,
```

Comment:
you want to pass `std::shared_ptr<const Ship>` not `const std::shared_ptr<Ship>`

Reply:
done.

Still doesnt work, always get back zero if I try to access Source properties (e.g. Source.MaxStructure). 

In the end I want the following to work in e.g. SR_JAWS
```(ShipPartMeter part = ThisPart meter = Capacity object = Source.ID)```

Reply:
It does actually work for normal ships. Just the temporary one used in the ship design does not work.

Reply:
Maybe it worked before (not sure). ThisPart was a culprit at least in the design screen. Using "SR_JAWS" instead worked.

Source.MaxStructure still does not work in the design screen but on in-game ships.

Reply:
Ok, that is because the initial value is zero. One needs to use `Value(Source.MaxStructure)`